src/freenet/client/async/SplitFileInserter.java

```diff
@@ -112,9 +112,13 @@ public void cancel(ClientContext context) {
     public void schedule(ClientContext context) throws InsertException {
         cb.onBlockSetFinished(this, context);
         storage.start();
+        SplitFileInserterSender s;
+        synchronized(this) {
+            s = sender;
+        }
         if(!ctx.getCHKOnly) {
-            sender.clearWakeupTime(context);
-            sender.schedule(context);
+            s.clearWakeupTime(context);
+            s.schedule(context);
```

Comment:
Are these safe functions thread-safe? It's not clear to me from reading them that they are. It'd be obviously correct if the lock was held for both these calls instead.